src/base/static/libs/maps/index.js

```diff
@@ -232,61 +236,96 @@ class MainMap {
   }
 
   geolocate() {
+    // TODO
     this.map.locate();
   }
 
-  addLayerView(collectionId) {
-    return model => {
-      this.layerViews[collectionId][model.cid] = new LayerView({
-        model: model,
-        router: this.router,
-        map: this.map,
-        layerGroup: this.layers[collectionId],
-        placeTypes: this.options.placeTypes,
-        // to access the filter
-        mapView: this,
-      });
+  createGeoJSONFromCollection(collection, modelIdToFocus) {
+    const geoJSON = {
+      type: "FeatureCollection",
+      features: [],
     };
-  }
 
-  removeLayerView(collectionId) {
-    return model => {
-      // remove map-bound events for this layer view
-      this.map.off(
-        "zoomend",
-        this.layerViews[collectionId][model.cid].updateLayer,
-        this.layerViews[collectionId][model.cid],
-      );
-      this.map.off(
-        "move",
-        this.layerViews[collectionId][model.cid].throttledRender,
-        this.layerViews[collectionId][model.cid],
-      );
-      this.map.off(
-        "zoomend",
-        this.layerViews[collectionId][model.cid].render,
-        this.layerViews[collectionId][model.cid],
-      );
+    collection.each(model => {
+      const properties = Object.keys(model.attributes)
+        .filter(key => key !== "geometry")
+        .reduce((geoJSONProperties, property) => {
+          geoJSONProperties[property] = model.get(property);
+          return geoJSONProperties;
+        }, {});
+
+      // We use "yes" and "no" here instead of booleans because I think it's
+      // a little easier to write filter expressions using string lookups.
+      properties.focused = modelIdToFocus === properties.id ? "yes" : "no";
```

Comment:
The comment in the code could have been clearer: when I tried implementing this with a boolean, I found that I had to explicitly declare the boolean within the filter expression using [`to-boolean`](https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-types-to-boolean).

So the whole filter expression would be:

```
idea:
 - filter:
     - "=="
     - - "get"
       - "isFocused"
     - -  "to-boolean"
       - true
```

It's another layer of array nesting, but I guess it's not really that bad. Looking back at this I think the additional nesting is preferable to an arbitrary convention like `yes`/`no`, so I'll change it.

Reply:
I'm still getting familiar with the Mapbox API, but I agree that using the `to-boolean` expression makes the most sense.

We can always form our own representation for the filters, but when/if we do, we should go all in. For example, we can have an actual Filters table that will contain the names of a few common filters, which we can then map to functions, like so:

```js
const isFocused = testIsFocused => {
  return [ "==", [ "get", "focused" ], [ "to-boolean", testIsFocused ] ]
}
```

We can even introduce this syntax into our YAML file now, eg:

```

 - filter:
     - isFocused:
         - true
```

although I haven't really thought it through, and not sure if it's worth the effort. It might make more sense to do this when we start porting the config to the db. But it's something worth considering.

Reply:
> We can always form our own representation for the filters

I was thinking about this a bit-- if we want to support different mapping providers at some point I think we'll need to create our own filter representation. I like the idea of functions for commonly used filters. Maybe we should create abstractions for common filters but also leave in the place the ability to write "native" filter expressions.

But I agree we could wait to do all that-- for now I think it's fine to just represent raw Mapbox-style filters in the config.